src/packet/aead_encrypted_data.js

```diff
@@ -64,17 +65,19 @@ class AEADEncryptedDataPacket {
   /**
    * Parse an encrypted payload of bytes in the order: version, IV, ciphertext (see specification)
    * @param {Uint8Array | ReadableStream<Uint8Array>} bytes
+   * @throws {Error} on parsing failure
    */
   async read(bytes) {
     await stream.parse(bytes, async reader => {
       const version = await reader.readByte();
       if (version !== VERSION) { // The only currently defined value is 1.
         throw new UnsupportedError(`Version ${version} of the AEAD-encrypted data packet is not supported.`);
       }
-      this.cipherAlgo = await reader.readByte();
-      this.aeadAlgo = await reader.readByte();
+      this.cipherAlgorithm = enums.write(enums.symmetric, await reader.readByte());
+      this.aeadAlgorithm = enums.write(enums.aead, await reader.readByte());
```

Comment:
I would prefer not to get into the habit of using `enums.write` for checking an integer value. I know it supports it, but I think we should get a bit more strict about the types, and use `read` only for `int` -> `string`, and `write` only for `string` -> `int`. And then we can do validation in some other way - for example, in this case we will already throw below, in the `read` call in `getAEADMode`, if the AEAD mode is unknown (and similarly for the cipher, but during decryption).

Reply:
Conceptually speaking, they're not necessarily "wrong enums", it might be that it's an algorithm that we just don't support. In other words, the spec does not forbid any values here, it's just that we might not be able to decrypt all of them. And sometimes it might be useful to parse the packet, e.g. when building a packet listing, and just show it with an "unsupported algorithm", rather than showing the entire packet as "unsupported packet".

> with potentially unexpected side effects

What side effects are you thinking of?

(There also shouldn't be a need for explicit exceptions, the "rule" would just be "throw when you use the enum value if we don't support it.")